### ISSUE TITLE
Find / create missing linked notes

### ISSUE THREAD

**Issue by ognjenio** [dendronhq/dendron#355]
**Is your feature request related to a problem? Please describe.**
Sometimes adding a link to a note before it exists is convenient. This is particular true for tags. While writing, I often add links to tags. Only after do I realize that they don't exist yet. This has resulted in many notes linking to tags that don't exist. 

To solve this I'd have to go through them one by one and click one each tag, and create missing files. 

**Describe the solution you'd like**
A way to more efficiently create linked notes -- there might be a few ways to do it:
1. When adding a link, if a note is not found, add a suggestion to create it
2. When running Doctor, create all notes that are linked but don't exist (or at least show a list of notes that are linked but don't exist)
3. Add a new command that can find all broken links

**Describe alternatives you've considered**
`grep` all `[[]]` and then looping through them in bash to create files.

**Comment by kevinslin** [comment]
Great suggestion. To start off, I think adding this to Doctor (2) makes sense. I also like 1. 

We can do either opt in or opt out via config (eg. create a file if link doesn't exist and then notify user that a file has been created - this would happen on file save to avoid creating spurious files). 

Would the opposite of this also be useful? (eg. delete empty notes that have been around for longer than one day)

**Comment by regoje** [comment]
I don't think immediately deleting would be good because it's destructive,
but having a search that lists all empty files would be very helpful.

Another extension accomplished functionality similar to search results by
opening a new file and dumping the search results there. I can't recall
which one at the moment though but it worked well.

**Comment by hikchoi** [comment]
I'm working my way towards adding a doctor command action `CREATE_MISSING_LINKED_NOTES` to start with. Once I get that to work, I'll look into adding a config to enable this function on save.

**Comment by chmac** [comment]
I think that this has been implemented, maybe this issue can be closed?

**Comment by aleksey-rowan** [comment]
You are correct, @chmac, 
- Doctor now has the [createMissingLinkedNotes](https://wiki.dendron.so/notes/ZeC74FYVECsf9bpyngVMU/#createmissinglinkednotes) command that will create notes that are linked to but don't exist,
- and [findBrokenLinks](https://wiki.dendron.so/notes/ZeC74FYVECsf9bpyngVMU/#findbrokenlinks) will show a list of broken links

In workspaces with two or more vaults the above commands only work on links the vault prefix though, if you have a multi-vault setup and links without the `dendron://` prefix, it won't work.

@ognjenio, if you would still like an option to show a suggestion to create a note when adding a link, please open a new issue for that.
